beamer/agent/chain.py

```diff
@@ -327,38 +327,47 @@ def fill_request(request: Request, context: Context) -> None:
             request_amount=request.amount,
         )
         return
-
-    if (
-        token.functions.allowance(context.address, context.fill_manager.address).call()
-        < request.amount
-    ):
-        func = token.functions.approve(context.fill_manager.address, allowance)
+    chain_id = ChainId(w3.eth.chain_id)
+    token_address = token.address
+    if (chain_id, token_address) not in context.fill_mutexes:
+        mutex = threading.Lock()
+        context.fill_mutexes.add((chain_id, token_address), mutex)
```

Comment:
This has not been addressed, I'm not sure whether github ate my comment, but anyway... we should create these locks during context creation, not here. Also, we should _not_ remove these locks -- they should be alive and reachable for as longer their contexts are.